### ISSUE TITLE
Issues with Google cast devices

### ISSUE THREAD

**Issue by erkr** [music-assistant/hass-music-assistant#713]
### What version of Music Assistant has the issue?

2022.7.0

### The problem

Since my upgrade to the final version, I observe significant more often casting  errors. Off course this can be an networking issue on my side, but I don’t observe other issues and believe it could be related to the last update.

### How to reproduce

Casting to a CC player

### Relevant log output

```
Logger: homeassistant.components.cast.media_player
Source: components/cast/media_player.py:395 
Integration: Google Cast (documentation, issues) 
First occurred: 19:37:25 (1 occurrences) 
Last logged: 19:37:25

Failed to cast media http://192.168.178.47:8095/media_player.audio57serre/next. Please make sure the URL is: Reachable from the cast device and either a publicly resolvable hostname or an IP address 
``` 
``` 
Logger: music_assistant.players.stream
Source: /usr/local/lib/python3.10/site-packages/music_assistant/controllers/streams.py:466 
First occurred: 18:39:19 (2 occurrences) 
Last logged: 19:37:36

Abort: client(s) did not connect within 10 seconds.
```

### What version of Home Assistant Core are your running

2022.7.6

### What type of installation are you running?

Home Assistant OS

### On what type of hardware are you running?

Raspberry Pi

**Comment by marcelveldt** [comment]
Cast is very sensitive to errors and the cast integration in HA even more. Sending more than one command at once and the whole socket connection drops off already. It's a real nightmare and it seems that things are getting worse. Other player platforms like Sonos, DLNA and slimproto do not have these issues at all so it must be something in the HA cast integration triggered by MA (or maybe the issue is there always but MA triggers it somehow). 

Today I ran some tests streaming media from a standalone controller casting audio to a cast group in Hires audio (just to stress it a bit more), worked like a champ. Starting the stream from HA (even without MA) caused the socket error from time to time and even worse a lot of hiccups in the audio. Bizarre thing is that at time of those hiccups a socket error pops up in HA as if the socket connection is locking up the cast device itself.

I think there's some flaw going on with the cast platform but I like to do some more tests and maybe even find the cure.
Music Assistant puts a lot more stress on the devices then starting some low quality online audio stream or spotify connect sesssion so I think it just triggers an issue already lingering in the background.

Strange thing is that for the past 3 years MA was a standalone project consuming pychromecast itself to control cast players and that worked perfectly fine after I discovered to treat the socket connection with great care.
I think I'm going to borrow some of my old code to interface with the cast speakers to see if that calms down the annoying issues.

**Comment by erkr** [comment]
Thanks Marcel. Only observation from my side is that the stability got worse with the changes made in final version. Have seen the errors already for ac longer time indeed, but they happened not very frequent

**Comment by marcelveldt** [comment]
I managed to find the trigger for the issues with cast but not the root cause. For now I've disabled a part of the code that triggers the issue (the deadlocking behavior) - that fix is available in the 2022.7.1 patch release.

I will look into the issue in more depth later but at least there's a temporary fix available.
Side effect for this temporary effect is that you can now no longer press next on a google device itself (or shout next track to the google assistant). 

Please test if 2022.7.1 gives more stability to Cast devices.

**Comment by erkr** [comment]
Testing as we speak. So keep the issue open for the chasing the root cause. Thanks for the work around 🥳

**Comment by marcelveldt** [comment]
Found the root cause for the connectivity issues with cast - fixed in 2022.7.2
